cms/djangoapps/contentstore/management/commands/set_advanced_fields_all_courses.py

```
@@ -0,0 +1,248 @@
+            print("=" * 80)
+            updated_data = update_course_advanced_settings(
+                course_module, revised_update_fields, user
+            )
+
+            print("=" * 80)
+
+            for field in revised_update_fields.keys():
+                print(f"Updated data for course advanced settings `{field}` = {updated_data[field]}")  # pylint: disable=line-too-long
+                print("=" * 80)
+
+        # All Courses
+        else:
+            courses = modulestore().get_courses()
+            course_ids = [x.id for x in courses]
+
+            failed_update_advanced_fields_courses = []
+
+            for course_id in course_ids[:limit_bulk_update_operation]:  # pylint: disable=redefined-argument-from-local
+                try:
+                    course_module = modulestore().get_course(CourseKey.from_string(str(course_id)))
+
+                    revised_update_fields = self._force_catalog_visibility_about(
+                        update_fields, course_module
+                    )
+
+                    print(f"Applying revised_update_fields = {revised_update_fields} to course {course_id} using `{user}` user.\n")  # pylint: disable=line-too-long
+                    updated_data = update_course_advanced_settings(
+                        course_module, revised_update_fields, user
+                    )
+
+                    print("=" * 80)
+
+                    for field in revised_update_fields.keys():
+                        print(f"Updated data for course advanced settings `{field}` = {updated_data[field]}")  # pylint: disable=line-too-long
+                        print("=" * 80)
+
+                except Exception as err:  # pylint: disable=broad-except
+                    failed_update_advanced_fields_courses.append(str(course_id))
+                    print("=" * 30 + f"> Oops, failed to update fields for {course_id}")
+                    print("Error:")
+                    print(err)
```

Replace these three print statements by: `logger.exception(err)`. I think that by default we should crash on error. Add a `-f/--force` command to ignore errors.